controllers/toolchainconfig/toolchainconfig_controller.go

```diff
@@ -60,12 +73,70 @@ func (r *Reconciler) Reconcile(request reconcile.Request) (reconcile.Result, err
 			// Request object not found, could have been deleted after reconcile request.
 			// Owned objects are automatically garbage collected. For additional cleanup logic use finalizers.
 			// Return and don't requeue
-			reqLogger.Error(err, "it looks like the ToolchainConfig resource with the name 'config' was removed - the cache will use the latest version of the resource")
+			reqLogger.Error(err, "it looks like the toolchainconfig resource was removed - the cache will use the latest version of the resource")
 			return reconcile.Result{}, nil
 		}
 		// Error reading the object - requeue the request.
-		return reconcile.Result{}, err
+		reqLogger.Error(err, "error getting the toolchainconfig resource")
+		return defaultReconcile, err
 	}
 	updateConfig(toolchainConfig)
-	return reconcile.Result{}, nil
+
+	// Sync member configs to member clusters
+	sync := synchronizer{
+		logger:         reqLogger,
+		getMembersFunc: r.GetMembersFunc,
+	}
+
+	if syncErrs := sync.syncMemberConfigs(toolchainConfig); len(syncErrs) > 0 {
+		for cluster, errMsg := range syncErrs {
+			err := fmt.Errorf(errMsg)
+			reqLogger.Error(err, "error syncing to member cluster '%s'", cluster)
+		}
+		return defaultReconcile, r.setSyncFailureStatusCondition(reqLogger, toolchainConfig, syncErrs)
+	}
+	return defaultReconcile, r.setCompleteStatusCondition(reqLogger, toolchainConfig)
```

Comment:
Just out of curiosity - what will happen when we will modify the `ToolchainConfig` for a few times (let say 3 times) between the deployments? How many reconciliations there will be? Always only one, or 4 of them? 
IIRC, there is only one reconciliation scheduled for one CR with the given name `config`, is that correct?

Reply:
So it's interesting, I tried out this scenario (3 updates) and this is what I noticed from the logs:

 20:43:33
 20:43:43
 20:43:46 Update #1
 20:43:53 
 20:43:55 Update #2
 20:44:03 
 20:44:03 Update #3
 20:44:13
 20:44:23
 20:44:33
...

So it appears that the updates are just additional reconciles but the original cadence of scheduled reconciles is maintained throughout. So yes, only one reconciliation schedule for one CR is what it seems to be.

Reply:
Great - so it confirms my expectation :-) - thanks for verifying it :+1: